libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetwork/Classes/Network/Action/CSFAction.m

```diff
@@ -541,8 +541,8 @@ - (id)contentFromData:(NSData*)data fromResponse:(NSHTTPURLResponse*)response er
         content = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingAllowFragments error:&jsonParseError];
     }
     
-    // If it's an error here, it's a basic parsing error.
-    if (jsonParseError && error) {
+    // If it's an error here, it's a basic parsing error and error is not 500 internal server error.
+    if (jsonParseError && error && response.statusCode != 500) {
```

Comment:
Agree with Kevin, we can make the change so that JSON is only parsed for 4xx error